web/app/components/document/sidebar/related-resources.ts

```diff
@@ -0,0 +1,459 @@
+      }
+
+      if (resources.externalLinks) {
+        this.relatedLinks = resources.externalLinks;
+      }
+
+      this.loadingHasFailed = false;
+    } catch (e: unknown) {
+      this.loadingHasFailed = true;
+    }
+  });
+
+  /**
+   * The task to animate a resource with a highlight.
+   * Called when a resource is added or edited.
+   * Temporarily adds a visual indicator of the changed element.
+   */
+  protected animateHighlight = restartableTask(
+    async (selector: string | number) => {
+      schedule("afterRender", async () => {
+        let target: HTMLElement | null = null;
+
+        if (typeof selector === "number") {
+          target = htmlElement(`#related-resource-${selector}`);
+        } else {
+          target = htmlElement(
+            `.related-resource${selector} .related-resource-link`
+          );
+        }
+
+        next(() => {
+          maybeScrollIntoView(
+            target as HTMLElement,
+            this.args.scrollContainer,
+            "getBoundingClientRect",
+            10
+          );
+        });
+
+        const highlight = document.createElement("div");
+        highlight.classList.add("highlight-affordance");
+        target.insertBefore(highlight, target.firstChild);
+
+        const fadeInAnimation = highlight.animate(
+          [{ opacity: 0 }, { opacity: 1 }],
+          { duration: 50 }
+        );
+
+        await timeout(Ember.testing ? 0 : 2000);
+
+        const fadeOutAnimation = highlight.animate(
+          [{ opacity: 1 }, { opacity: 0 }],
+          { duration: Ember.testing ? 50 : 400 }
+        );
+
+        try {
+          await fadeInAnimation.finished;
+          await fadeOutAnimation.finished;
+        } finally {
+          fadeInAnimation.cancel();
+          fadeOutAnimation.cancel();
+          highlight.remove();
+        }
+      });
+    }
+  );
+
+  /**
+   * The task to save the document's related resources.
+   * Creates a PUT request to the DB and conditionally triggers
+   * the resource-highlight animation.
+   */
+  protected saveRelatedResources = task(
+    async (
+      cachedDocuments,
+      cachedLinks,
+      elementSelectorToHighlight?: string | number
+    ) => {
+      if (elementSelectorToHighlight) {
+        void this.animateHighlight.perform(elementSelectorToHighlight);
+      }
+
+      try {
+        await this.fetchSvc.fetch(
+          `/api/v1/${this.args.documentIsDraft ? "drafts" : "documents"}/${
+            this.args.objectID
+          }/related-resources`,
+          {
+            method: "PUT",
+            body: JSON.stringify(this.formattedRelatedResources),
+            headers: {
+              "Content-Type": "application/json",
+            },
+          }
+        );
+      } catch (e: unknown) {
+        this.relatedLinks = cachedLinks;
+        this.relatedDocuments = cachedDocuments;
+
+        this.flashMessages.add({
+          title: "Unable to save",
```

Comment:
Nit: might be somewhat helpful for the user to differentiate between another save (doc?) action.
```suggestion
          title: "Unable to save related resources",
```

Reply:
I kept this generic so it could also apply to non "Related resources"-labeled fields, e.g., "RFC." I've changed it to "Unable to save resource" which might split the difference? What do you think